meta-lmp-bsp/dynamic-layers/stm-st-stm32mp/recipes-support/stm32-mfgtool-files/stm32-mfgtool-files/provision.sh.in

```diff
@@ -94,7 +94,7 @@ function usage {
 # [...]
 
 stm32prog_otp_refresh_values () {
-	otp_displ=$(STM32_Programmer_CLI -c port=USB1 -otp displ)
+	otp_displ=$(STM32_Programmer_CLI -c port=USB1 -otp displ || echo "STM32_Programmer_CLI -otp displ returned an error")
```

Comment:
I would change it to
```
otp_displ=$(STM32_Programmer_CLI -c port=USB1 -otp displ) || echo "STM32_Programmer_CLI -otp displ returned an error"
```
So that the error is shown to a user instead of appending to `otp_displ` variable.

Reply:
I have tested your proposal: it works with `bash` and `dash`. But I am not so sure it is want we want.

The purpose of the `otp_displ=$(...)` assignment is, in my view, to retrieve the information once, so that we can parse it and know the OTP state. In my view, a closed STM32MP15x with an unreadable OTP is something _perfectly normal and supported_. So I wanted something that just negates the code `1` returned by `STM32_Programmer_CLI`. I could have written something as `|| /usr/bin/true`, but I think we could be more explicit with a sort of comment that would be written in the `$otp_displ` content. Those who inspect it will know more about the OTP state. 
My intention was not to issue a warning, nothing is malfunctioning.
Your proposal will generate a visible warning when the STM32MP15x is closed. My code consider a closed STM32MP15x as a feature.

Reply:
Thanks, it looks reasonable.